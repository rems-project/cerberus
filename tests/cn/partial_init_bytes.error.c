int main()
{
    int x;
    int *p = &x;
    /*@ to_bytes W(p); @*/
    char *p_char = (char *)p;
    /*@ focus W<char>, 2u64; @*/
    p_char[2] = 0xff;
    /*@ from_bytes RW<int>(p); @*/
}
