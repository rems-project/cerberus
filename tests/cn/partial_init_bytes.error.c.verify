return code: 1
tests/cn/partial_init_bytes.error.c:5:9: warning: experimental keyword 'to_bytes' (use of experimental features is discouraged)
    /*@ to_bytes W(p); @*/
        ^~~~~~~~ 
tests/cn/partial_init_bytes.error.c:9:9: warning: experimental keyword 'from_bytes' (use of experimental features is discouraged)
    /*@ from_bytes RW<int>(p); @*/
        ^~~~~~~~~~ 
tests/cn/partial_init_bytes.error.c:7:9: warning: focus: index added, no effect on existing resources (yet).
    /*@ focus W<char>, 2u64; @*/
        ^~~~~~~~~~~~~~~~~~~~ 
[1/1]: main -- fail
tests/cn/partial_init_bytes.error.c:8:5: error: Missing resource for writing
    p_char[2] = 0xff;
    ~~~~~~~~~~^~~~~~ 
Resource needed: W<char>(&&x[(u64)2'i32])
State file: file:///tmp/state__partial_init_bytes.error.c__main.html
