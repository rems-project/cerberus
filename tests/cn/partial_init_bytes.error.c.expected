return code: 1
tests/cn/partial_init_bytes.error.c:5:9: warning: experimental keyword 'to_bytes' (use of experimental features is discouraged)
    /*@ to_bytes Block(p); @*/
        ^~~~~~~~ 
tests/cn/partial_init_bytes.error.c:9:9: warning: experimental keyword 'from_bytes' (use of experimental features is discouraged)
    /*@ from_bytes Owned<int>(p); @*/
        ^~~~~~~~~~ 
tests/cn/partial_init_bytes.error.c:7:9: warning: extract: index added, no resources (yet) extracted.
    /*@ extract Block<char>, 2u64; @*/
        ^~~~~~~~~~~~~~~~~~~~~~~~~~ 
[1/1]: main -- fail
tests/cn/partial_init_bytes.error.c:8:5: error: Missing resource for writing
    p_char[2] = 0xff;
    ~~~~~~~~~~^~~~~~ 
Resource needed: Block<char>(&&x[(u64)2'i32])
State file: file:///tmp/state__partial_init_bytes.error.c__main.html
